packages/api/src/external/carequality/command/cq-organization/create-or-update-cq-organization.ts

```diff
@@ -0,0 +1,113 @@
+import { AddressStrict } from "@metriport/core/domain/location-address";
+import { out } from "@metriport/core/util/log";
+import { capture } from "@metriport/core/util/notifications";
+import { errorToString } from "@metriport/shared";
+import { makeCarequalityManagementAPIFhir } from "../../api";
+import { CQDirectoryEntryData } from "../../cq-directory";
+import { CQOrgDetails, CQOrgDetailsWithUrls, getCqAddress, getCqOrgUrls } from "../../shared";
+import { getCqOrg } from "./get-cq-organization";
+import { getOrganizationFhirTemplate } from "./organization-template";
+import { parseCQOrganization } from "./parse-cq-organization";
+
+export type CreateOrUpdateCqOrganizationCmd = Omit<
+  CQOrgDetails,
+  "addressLine1" | "city" | "state" | "postalCode" | "lat" | "lon"
+> & {
+  cxId: string;
+  oid: string;
+  name: string;
+  address: AddressStrict;
+};
+
+/**
+ * Creates or updates a Carequality organization (it can be a Metriport Organization or Facility).
+ */
+export async function createOrUpdateCqOrganization(
+  cmd: CreateOrUpdateCqOrganizationCmd
+): Promise<CQDirectoryEntryData> {
+  const [cqOrg, orgDetailsWithUrls] = await Promise.all([
+    getCqOrg(cmd.oid),
+    cmdToCqOrgDetails(cmd),
+  ]);
+  if (cqOrg) return await updateCQOrganization(orgDetailsWithUrls);
+  return await createCQOrganization(orgDetailsWithUrls);
+}
+
+export async function cmdToCqOrgDetails(
+  cmd: CreateOrUpdateCqOrganizationCmd
+): Promise<CQOrgDetailsWithUrls> {
+  const { cxId, oid, name, address } = cmd;
+  const { coordinates, addressLine } = await getCqAddress({ cxId, address });
+  const orgDetailsWithUrls: CQOrgDetailsWithUrls = {
+    ...cmd,
+    ...getCqOrgUrls(),
+    oid,
+    name,
+    addressLine1: addressLine,
+    city: address.city,
+    state: address.state,
+    postalCode: address.zip,
+    lat: coordinates.lat.toString(),
+    lon: coordinates.lon.toString(),
+  };
+  return orgDetailsWithUrls;
+}
+
+async function updateCQOrganization(
+  orgDetails: CQOrgDetailsWithUrls
+): Promise<CQDirectoryEntryData> {
+  const { log, debug } = out(`CQ updateCQOrganization - CQ Org OID ${orgDetails.oid}`);
+  const carequalityOrg = getOrganizationFhirTemplate(orgDetails);
+  const cq = makeCarequalityManagementAPIFhir();
+
+  try {
+    const resp = await cq.updateOrganization({
+      org: carequalityOrg,
+      oid: orgDetails.oid,
+    });
+    debug(`resp updateOrganization: `, () => JSON.stringify(resp));
+    return parseCQOrganization(resp);
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    const extra = {
+      orgOid: orgDetails.oid,
+      context: `cq.org.update`,
+      carequalityOrg,
+      error,
+    };
+    if (error.response?.status === 404) {
+      capture.message("Got 404 while updating Org @ CQ, creating it", { extra });
```

Comment:
nit: if not logging the error, maybe it's good to leave a comment it's intentional?

Reply:
I think if it's important enough to send an event to Sentry, it's defs important enough to be logged.